chapters/03-package-setup.Rmd

```diff
@@ -1,37 +1,125 @@
-# Setting up an R package
+# Setting up an R package {#package-setup}
 
 - Describing what a package is and when or why to make one
+    - Use [available](https://github.com/r-lib/available) to make a name (naming is hard).
 - Using devtools, usethis for development
-- e.g. `use_devtools()` and `use_usethis()`.
+    - e.g. `use_devtools()` and `use_usethis()`.
 - Developing functions
 - Basic explanation of what a function is and its components
-- Actual process of taking code and converting it into a function will
-    be done in next chapter
 - Making and using datasets
 - Downloading zipf data at this point, using `usethis::use_data_raw()`, write the script to download the data and process into `data/` via `usethis::use_data()`.
 
+## Learning Objectives
 
+* Identify when to create an R package
+* Create a new R package using `usethis::create_package()`
+* Describe the purpose of the folders and files in a skeleton R package
+* Add data to an R package
+* Set global defaults to ease package development with devtools and usethis
 
-## Final exercise
+## About Zipf's Law
+
+We'll create a package (`zipfs`) to analyze a fascinating result in the field of
+quantitative linguistics. Zipf’s Law states that the second most common word in
+a body of text appears half as often as the most common, the third most common
+appears a third as often, and so on. To test whether Zipf’s Law holds for a
+collection of classic novels that are freely available from Project Gutenberg,
+we will write a software package that counts and analyzes the word frequency
+distribution in any arbitrary body of text.
 
-<!-- NOTE: Preferably something linked to working towards completing the project assignments, maybe to be used as a milestone? -->
+## Outline of action steps
 
-1. Use `create_package()` to create a package called FIXME.  
+* Why create an R package? 
 
-2. Edit the `DESCRIPTION` file to add:
+    * Text: Whenever you want to use data or functions over more than one project.
+    * Text: Advantages over sharing code and/or data files:
+        * Easy for other people to obtain and install
+        * Built-in documentation of functions and data
+        * Incorporate package-level meta data: authors, version, etc.
 
-    * yourself as the author, and
-    * a one sentence description of the package.
+* What is an R package?
 
-3. Add data to the package
+    * Text: Files that follow the R package conventions. usethis and devtools, 
+    designed to help you do this.  
+    * Exercise: Install a simple package from GitHub.
```

Comment:
and cran?

Reply:
also talk about importing versus suing `:::`?

Reply:
They should of course be very familiar with installing from CRAN already. Good point about `::`, added it to material